exonum/src/storage/proof_list_index/mod.rs

```diff
@@ -330,6 +347,18 @@ where
         self.get_branch(self.root_key()).unwrap_or_default()
     }
 
+    /// TBD
+    pub fn list_hash(&self) -> Hash {
+        let mut length_bytes = vec![0; 8];
+        LittleEndian::write_u64(&mut length_bytes, self.len());
+
+        HashStream::new()
```

Comment:
Did you test performance? After #1046, I'm inclined to believe that an implementation without `HashStream` (i.e., allocating an array with length `1 + 8 + HASH_LEN`, writing the parts of the hashed value to it and then hashing the array) could be faster.